packages/components/src/entities/CreateSamplesSubMenu.tsx

```diff
@@ -116,6 +116,18 @@ export const CreateSamplesSubMenu: FC<Props> = memo(props => {
     if (disabled) {
         return <DisableableMenuItem operationPermitted={false}>{menuText}</DisableableMenuItem>;
     }
+    let selectionNoun = undefined;
+    let selectionNounPlural = undefined;
+    if (!isSamples) {
+        if (sampleManagerIsPrimaryApp()) {
```

Comment:
Please try to hand through the module context rather than relying on the global module context when possible. We're trying to factor out/away all usages of the global pattern.

```ts
const { moduleContext } = useServerContext();

if (sampleManagerIsPrimaryApp(moduleContext)) {
    selectionNoun = 'source';
    selectionNounPlural = 'sources';
} else {
    selectionNoun = 'data';
    selectionNounPlural = 'data';
}
```